plugins/rule_engines/irods_rule_language/src/nre.reHelpers1.cpp

```diff
@@ -50,16 +48,17 @@ int myPID;
 int initializeReDebug( rsComm_t *svrComm ) {
     char condRead[NAME_LEN];
 
-    if ( svrComm == NULL ) {
+    if (svrComm == nullptr) {
         return 0;
     }
     if ( GlobalREDebugFlag != 4 ) {
         return 0;
     }
 
-    myPID = ( int ) getpid();
-    myHostName[0] = '\0';
-    gethostname( myHostName, MAX_NAME_LEN );
+    myPID = getpid();
+    std::memset(myHostName, 0, sizeof(myHostName));
+    const auto hostname = irods::get_server_property<std::string>(irods::KW_CFG_HOST);
+    hostname.copy(myHostName, sizeof(myHostName) - 1);
```

Comment:
why have both `myHostName` and `hostname` here... maybe i'm just not seeing it...?

Reply:
`get_server_property` is returning a `std::string`, and `myHostName` seems to be a global/static `char[]` into which that string needs to be copied. `myHostName` is used elsewhere, presumably/hopefully.

Reply:
Correct.

Reply:
sold.  resolving.